File: 0082 Remove Duplicates from Sorted List II/0082.cpp
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* start = new ListNode(0, head); 
        ListNode* prev = start;
        ListNode* cur = head;

        while (cur){
            if (cur->next == nullptr || cur->next->val != cur->val){
                prev = cur;
            } else {
                while (cur->next != nullptr && cur->next->val == cur->val) cur = cur->next;
                prev->next = cur->next;
            }
            cur = cur->next;
        }

        return start->next;
    }
};

File: 0082 Remove Duplicates from Sorted List II/0082.py
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        start = ListNode(0, head)
        prev, cur = start, head

        while cur:
            if not cur.next or cur.next.val != cur.val:
                prev = cur
            else:
                while cur.next and cur.next.val == cur.val:
                    cur = cur.next
                prev.next = cur.next
            cur = cur.next

        return start.next
